lib/provider/local/index.js

```diff
@@ -73,6 +75,32 @@ exports.validate = function(skew, request) {
   }
 };
 
+const digest =
+
+/**
+ * Validate correctness of the request's digest header
+ *
+ * @param  {String} algorithm The configured hashing algorithm
+ * @param  {HTTP.IncomingMessage} request The request to validate
+ *
+ * @throws {Errors.AuthorizationError}    If the digest header does not match the request body
+ */
+exports.digest = function(algorithm, request) {
+  const body = request.body.toString();
+  const digestHeader = request.headers.digest;
+
+  const hash = Crypto.createHash(algorithm);
+
+  hash.update(body);
```

Comment:
The [docs for Hash.update](https://nodejs.org/docs/latest-v4.x/api/crypto.html#crypto_hash_update_data_input_encoding) say the encoding defaults to `binary` if a string is provided without an encoding. Is that the right encoding, or should we explicitly set it to `utf8`?